openeo-geotrellis/src/main/scala/org/openeo/geotrellis/layers/FileLayerProvider.scala

```diff
@@ -1415,13 +1416,27 @@ class FileLayerProvider private(openSearch: OpenSearchClient, openSearchCollecti
        * Several edge cases to cover:
        *  - if feature extent is whole world, it may be invalid in target crs
        *  - if feature is in utm, target extent may be invalid in feature crs
-       *  this is why we take intersection
+       *  this is why we take intersection.
+       *  We convert both extents to a common CRS before taking the intersection.
+       *  If one of the CRSes can cover the whole world (non-UTM), this will be used as common CRS.
+       *  We give priority to use the target CRS as common one, because the intersection will be converted to it anyway
        */
-      val targetExtentInLatLon = targetExtent.reproject(feature.crs.get)
-      val featureExtentInLatLon = feature.rasterExtent.get.reproject(feature.crs.get,LatLng)
-
-      val intersection = featureExtentInLatLon.intersection(targetExtentInLatLon).map(_.buffer(1.0)).getOrElse(featureExtentInLatLon)
-      val tmp = expandToCellSize(intersection.reproject(LatLng, targetExtent.crs), theResolution)
+      val featureIsUTM = feature.crs.get.proj4jCrs.getProjection.getName == "utm"
+      val targetIsUTM = targetExtent.crs.proj4jCrs.getProjection.getName == "utm"
+      val commonCrs = if (!targetIsUTM) targetExtent.crs
+      else if (!featureIsUTM) feature.crs.get
+      else targetExtent.crs // Avoid conversion imprecision by intersecting directly in the target CRS
```

Comment:
Assuming a method 'isValidExtent(crs,extent:ProjectedExtent)' exists to check if a certain ProjectedExtent is valid in a certain crs exists:

```suggestion
      val commonCrs = if (isValidExtent(targetExtent.crs,featureExtent)) targetExtent.crs
      else if (isValidExtent(feature.crs,targetExtent)) feature.crs.get
      else LatLng
```

Reply:
Ah, I see. That should be safe indeed. The current implementation assumes that in a request UTM will always be valid when projected to something else, and that a random CRS can not safely be projected to UTM.

Side note: I found that UTM has a huge tolerance for extends that are horizontally outside of their bounds. There were successful jobs where the extent was multiple UTM zones from the correct one and still worked. So an UTM check for validity mostly looks at the vertical coordinates.

Reply:
The final version is still assuming that a target crs that is not UTM is always safe to use?
Please add a test where you use a local reference system like EPSG:3812 and a 'feature' with global bounds.

Reply:
The latest version does use the isValidExtent trick:
https://github.com/Open-EO/openeo-geotrellis-extensions/blob/9c2ed933a60d6c217675d0c4a333734b93c77270/openeo-geotrellis/src/main/scala/org/openeo/geotrellis/layers/FileLayerProvider.scala#L1441-L1443


readDataCubeWithOpensearchClientUTM has a big LatLon feature (-180.0014880952381, -59.99851190473005, 179.998511906399, 80.00148809523809) where an UTM region is requested. I just added readDataCubeWithOpensearchClientLAEA for good measure.